### PR TITLE
J2me admin report

### PR DESCRIPTION
**Pull request by emord**
http://manage.dimagi.com/default.asp?220943#1110293

@nickpell I think you're the only other person familiar with this code

cc: @biyeun

### CODE REVIEW

File: corehq/apps/es/filters.py [hunk 1]
```diff
@@ -119,3 +119,6 @@ def nested(path, filter_):
             "filter": filter_
         }
     }
+
+def regexp(field, regex):
```
Comment (dimagimon) [inline corehq/apps/es/filters.py]: E302 expected 2 blank lines, found 1

File: corehq/apps/es/forms.py [hunk 2]
```diff
@@ -72,3 +73,9 @@ def user_ids_handle_unknown(user_ids):
     else:
         user_filter = filters.missing('form.meta.userID')
     return user_filter
+
+def j2me_submissions(gt=None, gte=None, lt=None, lte=None):
```
Comment (dimagimon) [inline corehq/apps/es/forms.py]: E302 expected 2 blank lines, found 1

**Comment by nickpell** [comment]
looks good pending tests. this will require a domain preindex, right?

**Comment by emord** [comment]
Yeah. those are pretty quick though (looks like the last ones were <3 minutes)

**Comment by nickpell** [comment]
tests need updating:

```
FAIL: test_instantiate_all (testapps.test_pillowtop.tests.test_settings.PillowtopSettingsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/testapps/test_pillowtop/tests/test_settings.py", line 33, in test_instantiate_all
    self.assertEqual(expected_meta[pillow_config.name], _pillow_meta_from_config(pillow_config))
AssertionError: {u'es_type': u'hqdomain', u'name': u'DomainPillow', u'advertised_name': u'corehq [truncated]... != {'es_type': 'hqdomain', 'name': 'DomainPillow', 'advertised_name': 'corehq.pillo [truncated]...
- {u'advertised_name': u'corehq.pillows.domain.DomainPillow.test_hqdomains_20160308_1304.testhq',
?  -                   -                                                         ^    ^^
+ {'advertised_name': 'corehq.pillows.domain.DomainPillow.test_hqdomains_20160318_1339.testhq',
?                                                                              ^    ^^
-  u'change_feed_type': u'CouchChangeFeed',
?  -                    -
+  'change_feed_type': 'CouchChangeFeed',
-  u'checkpoint_id': u'corehq.pillows.domain.DomainPillow.test_hqdomains_20160308_1304.testhq',
?  -                 -                                                         ^    ^^
+  'checkpoint_id': 'corehq.pillows.domain.DomainPillow.test_hqdomains_20160318_1339.testhq',
?                                                                            ^    ^^
-  u'couch_filter': u'domain/domains_inclusive',
?  -                -
+  'couch_filter': 'domain/domains_inclusive',
-  u'couchdb_type': u'Database',
?  -                -
+  'couchdb_type': 'Database',
-  u'couchdb_uri': u'http://couch:5984/test_commcarehq__domains',
?  -               -
+  'couchdb_uri': 'http://couch:5984/test_commcarehq__domains',
-  u'document_class': u'Domain',
?  -                  -
+  'document_class': 'Domain',
-  u'es_alias': u'hqdomains',
?  -            -
+  'es_alias': 'hqdomains',
-  u'es_index': u'test_hqdomains_20160308_1304',
?  -            -                      ^    ^^
+  'es_index': 'test_hqdomains_20160318_1339',
?                                    ^    ^^
-  u'es_type': u'hqdomain',
?  -           -
+  'es_type': 'hqdomain',
-  u'extra_args': {},
?  -
+  'extra_args': {},
-  u'full_class_name': u'corehq.pillows.domain.DomainPillow',
?  -                   -
+  'full_class_name': 'corehq.pillows.domain.DomainPillow',
-  u'include_docs': True,
?  -
+  'include_docs': True,
-  u'name': u'DomainPillow',
?  -        -
+  'name': 'DomainPillow',
-  u'unique_id': u'test_hqdomains_20160308_1304'}
?  -             -                      ^    ^^
+  'unique_id': 'test_hqdomains_20160318_1339'}
?                                     ^    ^^
----------------------------------------------------------------------
Ran 13 tests in 6.020s
FAILED (failures=1)
```

**Comment by emord** [comment]
:white_check_mark:

**Pull request merged**